[NEW] A clearer, more readable return statement

sidiousvic opened the pull request:
.filter() doing the odd value filtering, .reduce() doing the reducing.
In any case, the current solution's use of .reverse() is unnecessary.

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

sidiousvic committed:
A clearer, more readable return statement

.filter() doing the odd value filtering, .reduce() doing the reducing.
In any case, the current solution's use of .reverse() is unnecessary.

Dhirendra24 commented on guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/sum-all-odd-fibonacci-numbers/index.md:
```diff
@@ -90,9 +90,7 @@ As you get the next odd one, don't forget to add it to a global variable that ca
 
         // Sum only the odd numbers and return the value
         // First, reverse the array to avoid starting acc with the first/greater number when it's even
```
Please update the comment & `Code Explanation` section as well.

sidiousvic committed:
A clearer, more readable return statement

Use .filter() to get the odd values in array.
Use .reduce() to sum.

Two separate tasks, one line, easier to read and understand.
Updated comments and Code Explanation accordingly.

ojeytonwilliams commented on guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/sum-all-odd-fibonacci-numbers/index.md:
```diff
@@ -88,11 +88,8 @@ As you get the next odd one, don't forget to add it to a global variable that ca
-        return arrFib.reverse().reduce((acc, curr) => {
-            return acc + curr * (curr % 2);
-        }, 0);
+        // We return the array, filtering and summing (reducing) odd numbers at once. 
```
```suggestion
        // We filter the array to get the odd numbers and reduce them to get their sum.
```
I wasn't sure about the original wording as it sounded like we were returning the array itself.

ojeytonwilliams reviewed (commented):
Other than that one wording quibble, it looks good to me!

RandellDawson commented:
@ojeytonwilliams This PR has gone stale.

ojeytonwilliams committed:
Clarify comment wording

ojeytonwilliams commented:
@sidiousvic Thank you for your contribution to the page! :+1:
We're happy to accept these changes, and look forward to future contributions. :memo:
